pkg/lib/packagekit.js

```diff
@@ -479,3 +521,213 @@ export function getBackendName() {
     return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
                 "Get", ["org.freedesktop.PackageKit", "BackendName"]);
 }
+
+/**
+ * Refresh PackageKit Cache
+ * @param {boolean} force - force refresh the cache (expensive)
+ * @param {*} progress_cb - progress callback
+ */
+export function refresh(force, progress_cb) {
+    force = force || false;
+    return cancellableTransaction("RefreshCache", [force], progress_cb);
+}
+
+/**
+ * Remove a package
+ *
+ * @param {?string[] | undefined} pkgnames - packages to remove
+ * @param {?() => void} progress_cb - progress callback
+ */
+export async function remove_packages(pkgnames, progress_cb) {
+    const ids = [];
+
+    const flags = Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED | Enum.FILTER_NOT_SOURCE;
+    await cancellableTransaction("Resolve", [flags, pkgnames], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => ids.push(package_id),
+                                 });
+
+    if (ids.length === 0)
+        return Promise.reject(new TransactionError("not-found", "Can't resolve packages(s)"));
+
+    return cancellableTransaction("RemovePackages", [0, ids, true, false], progress_cb);
+}
+
+/**
+ * @param {string[]} pkgnames - packages to install
+ * @param {?() => void} progress_cb - optional progress callback
+ */
+export async function install_packages(pkgnames, progress_cb) {
+    const flags = Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE | Enum.FILTER_NEWEST;
+    const packages_to_install = new Set(pkgnames);
+    const ids = [];
+
+    // TODO: just always resolve all states?
+    await cancellableTransaction("Resolve", [flags | Enum.FILTER_INSTALLED, pkgnames], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => {
+                                         const pkg = package_id.split(";")[0];
+                                         if (packages_to_install.has(pkg))
+                                             packages_to_install.delete(pkg);
+                                     },
+                                 });
+
+    // everything installed skip
+    if (packages_to_install.size === 0)
+        return Promise.resolve();
+
+    await cancellableTransaction("Resolve", [flags | Enum.FILTER_NOT_INSTALLED, Array.from(packages_to_install)], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => ids.push(package_id),
+                                 });
+
+    if (ids.length === 0)
+        return Promise.reject(new TransactionError("not-found", "Can't resolve package(s)"));
+    else
+        return cancellableTransaction("InstallPackages", [0, ids], progress_cb)
+                .catch(ex => {
+                    if (ex.code != Enum.ERROR_ALREADY_INSTALLED)
+                        return Promise.reject(ex);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-57ee25eb-20251113-101307-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)